e2e-tests/desktop/tests/credentials.spec.js

```diff
@@ -0,0 +1,310 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+import path from 'path';
+import { expect, test } from '../fixtures/baseTest.js';
+import * as boundaryHttp from '../../helpers/boundary-http.js';
+import { execSync } from 'child_process';
+import * as vaultCli from '../../helpers/vault-cli';
+
+let org;
+let targetWithBrokeredVaultCredentials;
+let targetWithInjectedCredentials;
+let targetWithBrokeredStaticCredentials;
+let sshPrivateKeyCredential;
+let jsonCredential;
+let credentialLibrary;
+const secretPolicyName = 'kv-policy';
+const boundaryPolicyName = 'boundary-controller';
+const secretsPath = 'e2e_secrets';
+const secretName = 'cred';
+const __dirname = import.meta.dirname;
+const boundaryControllerPath = path.resolve(
+  __dirname,
+  '../../admin/tests/fixtures/boundary-controller-policy.hcl',
+);
+const kvPolicyPath = path.resolve(
+  __dirname,
+  '../../admin/tests/fixtures/kv-policy.hcl',
+);
+
+test.beforeAll(async () => {
+  await vaultCli.checkVaultCli();
+});
+
+test.beforeEach(
+  async ({
+    request,
+    targetAddress,
+    targetPort,
+    sshUser,
+    sshKeyPath,
+    vaultAddr,
+    page,
+  }) => {
+    execSync(`vault policy delete ${secretPolicyName}`);
+    execSync(`vault policy delete ${boundaryPolicyName}`);
+    execSync(`vault secrets disable ${secretsPath}`);
+
+    await page.goto('/');
+
+    org = await boundaryHttp.createOrg(request);
+    const project = await boundaryHttp.createProject(request, org.id);
+
+    // Create host set
+    const hostCatalog = await boundaryHttp.createStaticHostCatalog(
+      request,
+      project.id,
+    );
+    const host = await boundaryHttp.createHost(request, {
+      hostCatalogId: hostCatalog.id,
+      address: targetAddress,
+    });
+
+    const hostSet = await boundaryHttp.createHostSet(request, hostCatalog.id);
+    await boundaryHttp.addHostToHostSet(request, {
+      hostSet,
+      hostIds: [host.id],
+    });
+
+    // Create a static credential store and key pair credential
+    const credentialStore = await boundaryHttp.createStaticCredentialStore(
+      request,
+      project.id,
+    );
+    sshPrivateKeyCredential = await boundaryHttp.createStaticCredentialKeyPair(
+      request,
+      {
+        credentialStoreId: credentialStore.id,
+        username: sshUser,
+        sshKeyPath,
+      },
+    );
+
+    // Create a static credential store and json credential
+    jsonCredential = await boundaryHttp.createStaticCredentialJson(request, {
+      credentialStoreId: credentialStore.id,
+      data: {
+        nested: {
+          key1: 'value1',
+          key2: 'value2',
```

Comment:
Can we also include characters that would normally be escaped here as well to confirm the behavior?